mlflow/tracing/clients/local.py

```diff
@@ -27,8 +27,41 @@ def __init__(self):
         queue_size = MLFLOW_TRACING_CLIENT_BUFFER_SIZE.get()
         self.queue = deque(maxlen=queue_size)
 
+        # used for displaying traces in IPython notebooks
+        self._prev_execution_count = -1
+
+    def _display_trace(self, trace: Trace):
+        """
+        Display the trace in an IPython notebook. If multiple
+        traces are generated in the same cell, only the last
+        one will be displayed.
+
+        This function is a no-op if not running in an IPython
+        environment.
+        """
+        try:
+            from IPython import get_ipython
+            from IPython.display import display
+
+            if get_ipython() is None:
+                return
+
+            # check the current exec count to know if the user is
+            # running the command in a new cell or not. this is
+            # useful because the user might generate multiple traces
+            # in a single cell, and we only want to display the last one.
+            current_exec_count = get_ipython().execution_count
+            if self._prev_execution_count != current_exec_count:
+                self._prev_execution_count = current_exec_count
+                self.display_handle = display(trace, display_id=True)
+            else:
+                self.display_handle.update(trace)
+        except Exception:
```

Comment:
(minor, we can do it as follow-up) we can split try catch for importing part and the rest, and show `log.warning()` the actual display errors raised from the latter.